### PR TITLE
bug(core): Recovery reporting was delayed for final 100% mark

### PR DESCRIPTION
**Pull request by vishramachandran**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

* Recovery reporting was not happening exactly when offset was reached. Fixed that.
* Added more consistent logging format for shards and partIds

### CODE REVIEW

File: coordinator/src/main/scala/filodb.coordinator/IngestionActor.scala [hunk 1]
```diff
@@ -265,9 +265,10 @@ private[filodb] final class IngestionActor(dataset: Dataset,
           off }
         .until(_ >= endOffset)
         // TODO: move this code to TimeSeriesShard itself.  Shard should control the thread
-        .lastL.runAsync(shardInstance.ingestSched)
+        .lastOptionL.runAsync(shardInstance.ingestSched)
```
Comment (velvia) [inline coordinator/src/main/scala/filodb.coordinator/IngestionActor.scala]: When would this ever return None?   Not too clear on semantics of this

File: core/src/main/scala/filodb.core/memstore/TimeSeriesMemStore.scala [hunk 2]
```diff
@@ -140,16 +140,22 @@ extends MemStore with StrictLogging {
...
                     reportingInterval: Long): Observable[Long] = {
     val shard = getShardE(dataset, shardNum)
-    val endOffset = checkpoints.values.max
     shard.setGroupWatermarks(checkpoints)
-    var targetOffset = checkpoints.values.min + reportingInterval
-    stream.map(shard.ingest(_)).collect {
-      case offset: Long if offset > targetOffset || offset >= endOffset =>
-        targetOffset += reportingInterval
-        offset
+    if (endOffset < startOffset) Observable.empty
```
Comment (velvia) [inline core/src/main/scala/filodb.core/memstore/TimeSeriesMemStore.scala]: You don't need to pass in endOffset and startOffset, they are calculatable from checkpoints.

File: coordinator/src/main/scala/filodb.coordinator/IngestionActor.scala (hunk 1 above)
Comment (vishramachandran) [reply coordinator/src/main/scala/filodb.coordinator/IngestionActor.scala]: See the other changes in commit. None would be returned when endOffset < startOffset.
This is required to eagerly move state from Recovering to Active if there is no data in kafka.

File: core/src/main/scala/filodb.core/memstore/TimeSeriesMemStore.scala (hunk 2 above)
Comment (vishramachandran) [reply core/src/main/scala/filodb.core/memstore/TimeSeriesMemStore.scala]: I know - I wanted one place where the calculation was done. Hence the parameter is being passed in.

**Pull request merged**